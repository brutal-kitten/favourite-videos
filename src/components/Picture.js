import React, { Component } from 'react';

class Picture extends Component {

  handleClick = (event) => {
    event.preventDefault();
    this.props.playVideo(this.props.id);
  }

  render() {
    return(
      <div className="pic grid-item" >
        <img src={this.props.thumbnails} alt="cover picture of video" onClick={(event) => this.handleClick(event)} />
      </div>
    )
  }
}

export default Picture;
