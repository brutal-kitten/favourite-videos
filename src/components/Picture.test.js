import React from 'react';
import ReactDOM from 'react-dom';
import Picture from './Picture.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('should display an image', () => {
  const tree = TestRenderer.create(
    <Picture />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe ('shoud call playVideo function after click', () => {
  const mockPlayVideo = jest.fn();
  const props = {
    playVideo: mockPlayVideo,
    id: "127456",
    thumbnails: 'some url'
  };

  const wrapper = shallow(<Picture {...props}/>);

  wrapper.find('img').simulate('click', {
    preventDefault: () => {},
  });


  it('should call playVideo function with 1 arqument - id', () => {
    expect(mockPlayVideo.mock.calls.length).toBe(1);
    expect(mockPlayVideo.mock.calls[0][0]).toBe("127456");
  });


  it('should have given image url', () =>  {
    expect(props.thumbnails).toBe('some url');
  })
})
