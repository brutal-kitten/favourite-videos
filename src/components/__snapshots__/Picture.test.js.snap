// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display an image 1`] = `
<div
  className="pic grid-item"
>
  <img
    alt="thumbnail"
    onClick={[Function]}
    tabIndex="0"
  />
</div>
`;
